### PR TITLE
feat: Added explicit mode.

### LINKED ISSUE #22: Explicit mode
Hi. Cool generator.
I am migrating to your generator from this - https://github.com/tom-englert/Equatable.Fody. It uses Fody and explicitly defined properties to implement IEquatable. I would like to see the same mode in this generator, in which you need to explicitly specify the necessary properties.
It would also be great to think about supporting IIncrementalGenerator

### PR DESCRIPTION
**Pull request by HavenDV**
Closes #22

### CODE REVIEW

**Comment by HavenDV** [comment]
For the tests to work, you need to check this PR first: https://github.com/diegofrata/Generator.Equals/pull/25

File: Generator.Equals.Runtime/Generator.Equals.Runtime.csproj [hunk 1]
```diff
@@ -4,6 +4,7 @@
         <TargetFramework>netstandard2.0</TargetFramework>
         <RootNamespace>Generator.Equals</RootNamespace>
         <IsPackable>true</IsPackable>
+        <LangVersion>8.0</LangVersion>
```
Comment (HavenDV) [inline Generator.Equals.Runtime/Generator.Equals.Runtime.csproj:7]: I don't quite understand your concerns. I am using the latest version of C# even in net4.0 and it works well. Some features like Range operators will only work in net core/net5/net6, but this is limited by the compiler

**Comment by diegofrata** [comment]
Only one newer feature was used in a single place. I wasn’t sure if this would impact older frameworks or not, as I hadn’t really tried it myself and was on a Mac. I wanted to get this released and removing was the easiest thing!I will look into raising the version of the language when I work on it next time.---- On Fri, 26 Aug 2022 02:39:29 +0100  Konstantin ***@***.***> wrote ---- 
@HavenDV commented on this pull request. 
 
 
 
In Generator.Equals.Runtime/Generator.Equals.Runtime.csproj: 
> @@ -4,6 +4,7 @@ 
         <TargetFramework>netstandard2.0</TargetFramework> 
         <RootNamespace>Generator.Equals</RootNamespace> 
         <IsPackable>true</IsPackable> 
+        <LangVersion>8.0</LangVersion> 
 
I don't quite understand your concerns. I am using the latest version of C# even in net4.0 and it works well. Some features like Range operators will only work in net core/net5/net6, but this is limited by the compiler

File: Generator.Equals.Runtime/Generator.Equals.Runtime.csproj (hunk 1 above)
Comment (HavenDV) [reply Generator.Equals.Runtime/Generator.Equals.Runtime.csproj:7]: It's just that I see another simple improvement that could improve the code - the file-scope namespace, which requires C# 10.

**Pull request merged**